ql/src/java/org/apache/hadoop/hive/ql/ddl/ShowUtils.java

```diff
@@ -0,0 +1,397 @@
+        value.replaceAll("\\\\n|\\\\r|\\\\r\\\\n", "\n") : value;
+    formatOutput(name, unescapedValue, tableInfo);
+  }
+
+  /**
+   * Indent processing for multi-line values.
+   * Values should be indented the same amount on each line.
+   * If the first line comment starts indented by k, the following line comments should also be indented by k.
+   * 
+   * @param value the value to write
+   * @param tableInfo the buffer to write to
+   * @param columnWidths the widths of the previous columns
+   * @param printNull print null as a string, or do not print anything
+   */
+  private static void indentMultilineValue(String value, StringBuilder tableInfo, int[] columnWidths,
+      boolean printNull) {
+    if (value == null) {
+      if (printNull) {
+        tableInfo.append(String.format("%-" + ALIGNMENT + "s", value));
+      }
+      tableInfo.append(LINE_DELIM);
+    } else {
+      String[] valueSegments = value.split("\n|\r|\r\n");
+      tableInfo.append(String.format("%-" + ALIGNMENT + "s", valueSegments[0])).append(LINE_DELIM);
+      for (int i = 1; i < valueSegments.length; i++) {
+        printPadding(tableInfo, columnWidths);
+        tableInfo.append(String.format("%-" + ALIGNMENT + "s", valueSegments[i])).append(LINE_DELIM);
+      }
+    }
+  }
+
+  /**
+   * Print the rigth padding, with the given column widths.
+   * 
+   * @param tableInfo The buffer to write to
+   * @param columnWidths The column widths
+   */
+  private static void printPadding(StringBuilder tableInfo, int[] columnWidths) {
+    for (int columnWidth : columnWidths) {
+      if (columnWidth == 0) {
+        tableInfo.append(FIELD_DELIM);
+      } else {
+        tableInfo.append(String.format("%" + columnWidth + "s" + FIELD_DELIM, ""));
+      }
+    }
+  }
+
+  /**
+   * Helps to format tables in SHOW ... command outputs.
+   */
+  public static class TextMetaDataTable {
+    private List<List<String>> table = new ArrayList<>();
+
+    public void addRow(String... values) {
+      table.add(Lists.<String> newArrayList(values));
+    }
+
+    public String renderTable(boolean isOutputPadded) {
+      StringBuilder stringBuilder = new StringBuilder();
+      for (List<String> row : table) {
+        formatOutput(row.toArray(new String[] {}), stringBuilder, isOutputPadded, isOutputPadded);
+      }
+      return stringBuilder.toString();
+    }
+
+    public void transpose() {
+      if (table.size() == 0) {
+        return;
+      }
+      List<List<String>> newTable = new ArrayList<List<String>>();
+      for (int i = 0; i < table.get(0).size(); i++) {
+        newTable.add(new ArrayList<>());
+      }
+      for (List<String> sourceRow : table) {
+        if (newTable.size() != sourceRow.size()) {
+          throw new RuntimeException("invalid table size");
+        }
+        for (int i = 0; i < sourceRow.size(); i++) {
+          newTable.get(i).add(sourceRow.get(i));
+        }
+      }
+      table = newTable;
+    }
```

Comment:
There's got to be a better way of doing this...

List#addAll or something other than 1-by-1 iteration.

Reply:
I don't think it is possible in any other way. Transposing a table must be done like this in essence.